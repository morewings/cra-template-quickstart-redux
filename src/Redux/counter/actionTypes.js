export const INCREMENT_COUNTER = 'INCREMENT_COUNTER';
