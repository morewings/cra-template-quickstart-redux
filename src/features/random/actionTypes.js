export const GET_RANDOM_NUMBER = 'GET_RANDOM_NUMBER';
