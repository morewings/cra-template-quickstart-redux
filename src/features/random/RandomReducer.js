import {GET_RANDOM_NUMBER} from './actionTypes';

const initialState = {
  number: undefined,
  isLoading: false,
  hasError: false,
  isFulfilled: false,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case `${GET_RANDOM_NUMBER}_PENDING`:
      return {
        ...state,
        isFulfilled: false,
        isLoading: true,
        hasError: false,
        number: undefined,
      };

    case `${GET_RANDOM_NUMBER}_FULFILLED`:
      return {
        isFulfilled: true,
        isLoading: false,
        hasError: false,
        number: action.payload.data,
      };

    case `${GET_RANDOM_NUMBER}_REJECTED`:
      return {
        isFulfilled: false,
        isLoading: false,
        hasError: true,
        number: undefined,
      };

    default:
      return state;
  }
};
