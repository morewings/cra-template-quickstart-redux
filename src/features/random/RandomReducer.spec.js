import {GET_RANDOM_NUMBER} from './actionTypes';
import RandomReducer from './RandomReducer';

describe('features > random > RandomReducer', () => {
  it('returns initial state, if non matched action is dispatched', () => {
    const initialState = {
      isLoading: false,
      hasError: false,
      isFulfilled: false,
    };

    const action = {
      type: 'FOO',
    };

    expect(RandomReducer(initialState, action)).toBe(initialState);
  });

  /**
   * Provide table of values to run test case against
   * @see https://jestjs.io/docs/en/api#testeachtablename-fn-timeout
   */
  it.each([
    [`${GET_RANDOM_NUMBER}_FULFILLED`],
    [`${GET_RANDOM_NUMBER}_PENDING`],
    [`${GET_RANDOM_NUMBER}_REJECTED`],
  ])(`updates state according to dispatched action`, actionType => {
    const initialState = {
      value: 0,
    };

    const payload =
      actionType === `${GET_RANDOM_NUMBER}_FULFILLED`
        ? {
            data: 1,
          }
        : undefined;

    const action = {
      type: actionType,
      payload,
    };

    expect(RandomReducer(initialState, action)).toMatchSnapshot();
  });
});
