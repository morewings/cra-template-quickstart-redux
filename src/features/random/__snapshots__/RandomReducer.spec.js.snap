// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features > random > RandomReducer updates state according to dispatched action 1`] = `
Object {
  "hasError": false,
  "isFulfilled": true,
  "isLoading": false,
  "number": 1,
}
`;

exports[`features > random > RandomReducer updates state according to dispatched action 2`] = `
Object {
  "hasError": false,
  "isFulfilled": false,
  "isLoading": true,
  "number": undefined,
  "value": 0,
}
`;

exports[`features > random > RandomReducer updates state according to dispatched action 3`] = `
Object {
  "hasError": true,
  "isFulfilled": false,
  "isLoading": false,
  "number": undefined,
}
`;
